src/Resources/contao/library/Contao/User.php

```diff
@@ -502,19 +496,8 @@ public static function loadUserByUsername($username)
 		// Check if a passwords needs rehashing (see contao/core#8820)
 		if ($isLogin)
 		{
-			$blnNeedsRehash = true;
-
-			// Handle old sha1() passwords with an optional salt
-			if (preg_match('/^[a-f0-9]{40}(:[a-f0-9]{23})?$/', $user->password))
-			{
-				list($strPassword, $strSalt) = explode(':', $user->password);
-				$blnAuthenticated = ($strPassword === sha1($strSalt . $request->request->get('password')));
-			}
-			else
-			{
-				$blnAuthenticated = password_verify($request->request->get('password'), $user->password);
-				$blnNeedsRehash = password_needs_rehash($user->password, PASSWORD_DEFAULT);
-			}
+			$blnAuthenticated = password_verify($request->request->get('password'), $user->password);
+			$blnNeedsRehash = password_needs_rehash($user->password, PASSWORD_DEFAULT);
 
 			// Re-hash the password if the algorithm has changed
 			if ($blnAuthenticated && $blnNeedsRehash)
```

Comment:
Only partially related to this PR, but why does the User object rehash the password? That seems very wrong to me… should be a security component…

Reply:
It's still wrong in this place, just because we can use the request…

Reply:
FYI: https://github.com/symfony/symfony/pull/26050